beaker_kernel/contexts/oceananigans/context.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+import os
+from typing import TYPE_CHECKING, Any, Dict
+
+import requests
+
+from beaker_kernel.lib.context import BaseContext
+from beaker_kernel.lib.jupyter_kernel_proxy import JupyterMessage
+
+from .agent import OceananigansAgent
+
+if TYPE_CHECKING:
+    from beaker_kernel.kernel import LLMKernel
+    from beaker_kernel.lib.subkernels.base import BaseSubkernel
+
+
+logger = logging.getLogger(__name__)
+
+
+class OceananigansContext(BaseContext):
+
+    slug = "oceananigans"
+    agent_cls = OceananigansAgent
+
+    def __init__(self, beaker_kernel: "LLMKernel", subkernel: "BaseSubkernel", config: Dict[str, Any]) -> None:
+        self.target = "oceananigan"
+        self.intercepts = {
+            "save_data_request": (self.save_data, "shell"),
+        }
```

Comment:
The @intercept decorator has been merged in upstream since this was written, let's switch over to using that for consistency.
Keep in mind that the signature of the `save_data` method will change a little bit.